subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -231,51 +285,38 @@ static uint16_t *scene_find(struct bt_mesh_scene_srv *srv, uint16_t scene)
 static void entry_recover(struct bt_mesh_scene_srv *srv, bool vnd,
 			  const struct scene_data *data)
 {
-	sys_slist_t *list = vnd ? &srv->vnd : &srv->sig;
-	struct bt_mesh_scene_entry *entry;
-
-	SYS_SLIST_FOR_EACH_CONTAINER(list, entry, n) {
-		if (data->elem_idx != entry->model->elem_idx) {
-			continue;
-		}
-
-		/* MeshMDL1.0.1, section 5.1.3.1.1:
-		 * If a model is extending another model, the extending model shall determine
-		 * the Stored with Scene behavior of that model.
-		 */
-		if (bt_mesh_model_is_extended(entry->model)) {
-			continue;
-		}
-
-		if (vnd) {
-			uint16_t company_id = sys_get_le16(data->data);
-
-			if (entry->model->vnd.id != data->id) {
-				continue;
-			}
+	const struct bt_mesh_elem *elem = &bt_mesh_comp_get()->elem[data->elem_idx];
+	const size_t overhead = vnd ? VND_MODEL_SCENE_DATA_OVERHEAD : 0;
+	const struct bt_mesh_scene_entry *entry;
+	struct bt_mesh_model *mod;
```

Comment:
This just a question. Was it not a PR earlier where all "mod" was changed to "model"? If this is not true, disregard this.